AliakseiLiauchuk/src/containers/Users.js

```diff
@@ -0,0 +1,34 @@
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import { getUsers, addUser } from '../actions/UserActions'
+import { UsersList } from '../components/UsersList'
+
+class Users extends Component {
+  addUser() {
+    this.props.dispatch(addUser());
+  }
+
+  componentDidMount() {
+    const { dispatch } = this.props
+    dispatch(getUsers())
+  }
+
+  render() {
+    const { users } = this.props
+    return (<>
+              <button type="submit" onClick={this.addUser.bind(this)}>add user</button>
+              <UsersList users={users} />
+            </>
+    )
+  }
+}
+
+const mapStateToProps = state => {
+  const { usersAll } = state
+
+  return {
+    users: usersAll.items
+  }
+}
+
+export default connect(mapStateToProps)(Users)
```

Comment:
`import React, { Component } from 'react'
import { connect } from 'react-redux'
import { fetchUsers, fetchUser } from '../actions'
import { UsersList } from '../components/UsersList'

class Users extends Component {
  fetchUser(user) {
    this.props.dispatch(fetchUser(user));
  }

  componentDidMount() {
    this.props.dispatch(fetchUsers())
  }

  render() {
    return (<>
              <button type="submit" 
              onClick={this.fetchUser.bind(this, 
                {username: "newUsername",
                  name: "newName",
                  email: "newEmail",
                  phone: "newPhone"
                })}>add user</button>
              <UsersList users={this.props.users} />
            </>
    )
  }
}

const mapStateToProps = state => {
  return {
    users: state
  }
}

export default connect(mapStateToProps)(Users)
`